src/main/java/bio/terra/workspace/app/controller/common/ReferenceController.java

```diff
@@ -0,0 +1,131 @@
+        requestMetadata.toString());
+    DataReferenceValidationUtils.validateReferenceName(requestMetadata.getName());
+    dataReferenceValidation.validateReferenceObject(referenceObject, referenceType, userReq);
+
+    DataReferenceRequest referenceRequest =
+        DataReferenceRequest.builder()
+            .workspaceId(workspaceId)
+            .name(requestMetadata.getName())
+            .referenceDescription(requestMetadata.getReferenceDescription())
+            .referenceType(referenceType)
+            .cloningInstructions(
+                CloningInstructions.fromApiModel(requestMetadata.getCloningInstructions()))
+            .referenceObject(referenceObject)
+            .build();
+    DataReference reference = dataReferenceService.createDataReference(referenceRequest, userReq);
+
+    logger.info(
+        "Created data reference {} in workspace {} for {}",
+        reference.toString(),
+        workspaceId.toString(),
+        userReq.getEmail());
+    return reference;
+  }
+
+  protected DataReference getReference(
+      UUID workspaceId,
+      UUID referenceId,
+      AuthenticatedUserRequest userReq,
+      DataReferenceService dataReferenceService) {
+    logger.info(
+        "Getting data reference by id {} in workspace {} for {}",
+        referenceId.toString(),
+        workspaceId.toString(),
+        userReq.getEmail());
+    DataReference ref = dataReferenceService.getDataReference(workspaceId, referenceId, userReq);
+    logger.info(
+        "Got data reference {} in workspace {} for {}",
+        ref.toString(),
+        workspaceId.toString(),
+        userReq.getEmail());
+    return ref;
+  }
+
+  protected DataReference getReferenceByName(
+      UUID workspaceId,
+      DataReferenceType type,
+      String name,
+      AuthenticatedUserRequest userReq,
+      DataReferenceService dataReferenceService) {
+    logger.info(
+        "Getting data reference by name {} in workspace {} for {}",
+        name,
+        workspaceId.toString(),
+        userReq.getEmail());
+    DataReferenceValidationUtils.validateReferenceName(name);
+    DataReference ref =
+        dataReferenceService.getDataReferenceByName(workspaceId, type, name, userReq);
+    logger.info(
+        "Got data reference by name {} in workspace {} for {}",
+        ref.toString(),
+        workspaceId.toString(),
+        userReq.getEmail());
+    return ref;
+  }
+
+  protected void updateReference(
+      UUID workspaceId,
+      UUID referenceId,
+      UpdateDataReferenceRequestBody body,
+      AuthenticatedUserRequest userReq,
+      DataReferenceService dataReferenceService) {
+    if (body.getName() == null && body.getReferenceDescription() == null) {
+      throw new InvalidDataReferenceException(
+          "Must specify name or referenceDescription to update.");
+    }
+
+    if (body.getName() != null) {
+      DataReferenceValidationUtils.validateReferenceName(body.getName());
+    }
+
+    logger.info(
+        String.format(
```

Comment:
You don't need the almost all the time. You can use placeholders {} in the log message.

Reply:
Ah, I missed this one. thanks!